Add support for hash: ignore

jmelis opened the pull request.

jmelis committed:
Add support for hash: ignore

jmelis commented:
@riuvshin @jfchevrette can you review?

I marked it as WIP because I still want to add some tests for this, but the functionality should be ready.

riuvshin commented on saasherder/saasherder.py:
```diff
@@ -152,13 +152,44 @@ def mkdir_templates_dir(self):
+        """
+
+        if service_hash == 'ignore':
+            return 'master'
```
does this works for both GL and GH?

riuvshin reviewed (approved):
looks good!

jmelis committed:
Add clean target to Makefile

jmelis committed:
Change test templates to saas-openshiftio

The `saas` repo is no longer used. Additionally the templates it points
to don't have the `IMAGE` or `IMAGE_TAG` params.

jmelis committed:
Add test for the hash: ignore functionality

jmelis committed:
commit the redirector-ignore template

jmelis commented:
@jfchevrette this is ready for merge, but can you please review first?

jmelis committed:
formatting

jmelis committed:
more formatting

riuvshin commented:
@jmelis can we have at least a unit test that will test `raw_gitlab()` ? we can't have e2e becuase of vpn but maybe we can mock it

jmelis commented:
@riuvshin that is a good idea, but it would take some effort to do so

jfchevrette commented:
LGTM! Thanks a lot @jmelis
